src/RFC-0314-VNCSelection.md

```diff
@@ -0,0 +1,88 @@
+## Goals
+The goal of this RFC is to describe the process for allocating Validator Nodes (VNs) to Validator Node Committees (VNCs)
+
+## Intro
+Validator nodes will have to group themselves into Validator Node Committees for transactions/shard processing. Committees will be formed for each new transaction/shard processing. This committee needs to be determined randomly, and we use the VN_Key as this changes periodically and is random. Each VN will be allocated a unique and individual shard space to serve in as a VNC member. 
+
+## Committee Creation
+Because we have the base layer where each VN needs to publish a registration transaction, we can get base_node and miners to keep track of all active VNs. We represent all VNs in a (balanced) Merkle tree with the VN_keys as leaves. We declare a constant `COMMITTEE_SIZE` which can be changed in the consensus constants.
+When a transaction is proposed, the shards for each substate determine the VN committee from the Merkle tree by finding the `COMMITTEE_SIZE / 2` VN_keys to the 
+left and `COMMITTEE_SIZE / 2` VN_Keys to the right in the Merkle tree.
+
+This makes it very easy to determine what shard space a VN needs to serve, which states the VN needs to sync from peers, and who has them. Because the
+second layer has a delayed view of the network. VNs can also know beforehand and prep to ensure they are ready when new block heights appear. 
+
+## Committee proofs
+We construct the balanced Merkle tree from all active VN registration transactions and prune away all inactive ones. Because the [base node]s have to keep track
+of the entire unspent UTXO set, it becomes easy for them to track and validate all active VN registration UTXOs. This means we can keep [base node]s 
+responsible for validating and constructing the Merkle tree. We commit this Merkle tree per block as a Merkle root inside the block's header as the `validator_node_set_root`. 
+
+This Merkle root in the header always lags by one block, meaning that the Merkle root is for the state of the VN's before the start of the block it's mined in. 
+When a VN registers, that new VN key will only appear inside the Merkle tree in the next block. The reason for this is that header_hash is used in the
+the calculation process of the VN_key. 
+
+### Option 2 for Committee proofs
+The VN_key needs some random entropy that's not minable to ensure that a VN cannot choose its VN_key. This random entropy is currently the block hash of the
+block the VN registration was mined in. Hence the reason for the Merkle root lagging by one block. If we change this entropy value be another source of 
+randomness, such as the `utxo_merkle_root`, then we don't have to do the lag by 1, and it can all be calculated in the single block.
+
+## Leader selection
+The VNC leader should not be decided beforehand and must be chosen randomly only when a tx is published. We also need to select an order of leaders
+in the case, the first leader is offline/and or not responsive.

```

Comment:
Every instance of the word "random" in this RFC is misleading. Everything must be deterministic, otherwise, there can be no consensus.